pkg/driver/controllerserver.go

```diff
@@ -164,6 +165,24 @@ func (cs *controllerServer) CreateVolume(_ context.Context, req *csi.CreateVolum
 		bucketName = secretMapCustom["bucketName"]
 	}
 
+	// Check for BucketVersioning parameter
+	if val, ok := secretMap["BucketVersioning"]; ok && val != "" {
+		enable := strings.ToLower(strings.TrimSpace(val))
+		if enable != "true" && enable != "false" {
+			return nil, status.Error(codes.InvalidArgument, fmt.Sprintf("Invalid BucketVersioning value in secret: %s. Must be 'true' or 'false'", val))
+		}
+		BucketVersioning = enable
+		klog.Infof("BucketVersioning set via secret: %s", BucketVersioning)
```

Comment:
klog.Infof("BucketVersioning value that will be set via secret: %s", BucketVersioning)